components/renku_data_services/secrets/models.py

```diff
@@ -1,35 +1,58 @@
 """Base models for secrets."""
 
-from dataclasses import dataclass
-from datetime import UTC, datetime
-from enum import Enum
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import StrEnum
 
 from kubernetes import client as k8s_client
-from pydantic import BaseModel, Field
 from ulid import ULID
 
 
-class SecretKind(Enum):
+class SecretKind(StrEnum):
     """Kind of secret. This should have the same values as users.apispec.SecretKind."""
 
     general = "general"
     storage = "storage"
 
 
-class UnsavedSecret(BaseModel):
+@dataclass(frozen=True, eq=True, kw_only=True)
+class UnsavedSecret:
```

Comment:
issue: This is used only as the base class for `Secret` below. So it can be eliminated. I was confused why we have both `UnsavedSecret` and `NewSecret`. 

Also in most other components we use `Unsaved...` rather than `New...` so it is nicer if we kept the same logic here. Which means eliminate `NewSecret` and reaplace it with `UnsavedSecret`. And have `Secret` not inherit from `UnsavedSecret`

Reply:
Done in 27f2c356df009150f6b07c176a42e70263d5ab77.